Runtime/Interaction/CustomMeshUtils/VertexRemapper.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ubco.ovilab.HPUI.CustomMeshUtils
+{
+    public static class VertexRemapper 
+    {
+        /// <summary>
+        /// For a given rectangular mesh, returns an integer array with remapped vertices from bottom left to top right
+        /// </summary>
+        /// <param name="remapData">Scriptable object to store the remapped indices</param>
+        /// <param name="mesh">The mesh to sort vertices for</param>
+        /// <param name="smesh">The skinned mesh renderer to extract the mesh from</param>
+        /// <returns></returns>
+        public static void GetRectifiedIndices(VertexRemapData remapData, Mesh mesh)
```

Comment:
This is only used in once place. Doesn't need to be public nor in a separate class. Please fold this implementation into the HPUICustomMesh class.